### ISSUE TITLE
Wrong figure caption

### ISSUE THREAD

**Issue by psychelzh** [yihui/hugo-xmin#22]
I tried to use `.Rmd` to generate figures with captions using R code chunks.

Following lines are used

````r
```{r load data and visualization, message=FALSE, warning=FALSE, fig.cap=c("Matrix plot of all variables", "Linear trend of different groups")}
library(tidyverse)
birthsmoke <- read.table("https://onlinecourses.science.psu.edu/stat501/sites/onlinecourses.science.psu.edu.stat501/files/data/birthsmokers.txt", header = TRUE)
plot(birthsmoke)
ggplot(birthsmoke, aes(Gest, Wgt, color = Smoke)) +
  geom_point(shape = 1) +
  geom_smooth(method = "glm", se = FALSE) +
  theme_minimal()
```
````

But unfortunately, the result is not intended (following screenshot). Is it possible to support caption more properly?

![image](https://user-images.githubusercontent.com/13306020/35484138-cbeb787e-0486-11e8-9130-8dadab96ed22.png)

**Comment by yihui** [comment]
See https://bookdown.org/yihui/bookdown/figures.html for the requirements on chunk labels.

**Comment by psychelzh** [comment]
I have read those texts. I made a simpler example. But the caption is still not right (with unnecessary **(#fig:load_data_and_visualization)** heading).

````r
```{r load_data_and_visualization, message=FALSE, warning=FALSE, fig.cap="Matrix plot of all variables"}
library(tidyverse)
birthsmoke <- read.table("https://onlinecourses.science.psu.edu/stat501/sites/onlinecourses.science.psu.edu.stat501/files/data/birthsmokers.txt", header = TRUE)
plot(birthsmoke)
```
````

![image](https://user-images.githubusercontent.com/13306020/35548398-1432317a-05ba-11e8-9753-5d6e4acc5676.png)

**Comment by yihui** [comment]
You didn't read it carefully. To quote the documentation on that page:

> If you want to cross-reference figures or tables generated from a code chunk, please make sure the chunk label only contains alphanumeric characters (a-z, A-Z, 0-9), slashes (/), or dashes (-).

**Comment by psychelzh** [comment]
I have changed it without spaces now. You could read my second example.

**Comment by psychelzh** [comment]
The point here is that the words **"(#fig:load_data_and_visualization)"** are unnecessary. I do not know why they appear here.

**Comment by yihui** [comment]
Okay, for the third time: please, please read the documentation carefully. I have copied the relevant part above. It is only one sentence. Please read every word in it.

**Comment by psychelzh** [comment]
Oh my, I know now. The underscore (`_`) is not supported. :( I am sorry, but it is very error-prone for newbies if underscore is not supported.

**Comment by psychelzh** [comment]
I learned more about `HTML` these days, and I found this description on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id):

> Note: Using characters except ASCII letters, digits, '_', '-' and '.' may cause compatibility problems, as they weren't allowed in HTML 4. Though this restriction has been lifted in HTML 5, an ID should start with a letter for compatibility.

And I know why underscore is not supported now. It is because `blogdown` will use the chunk label as the value of `id` attribute of the `img` element that, just as that quote says, no underscores/spaces are permitted. Thank you for your time on this. 👍
